File: examples/javascript/Ultra/UltraWebApplicationWithAssets1/UltraWebApplicationWithAssets1/UltraApplication.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Web;
using java.applet;
using java.awt;
using ScriptCoreLib.ActionScript.Extensions;
using ScriptCoreLib.ActionScript.flash.display;
using ScriptCoreLib.JavaScript.DOM.HTML;
using ScriptCoreLib.JavaScript.Extensions;
using ScriptCoreLib.Ultra.Library.Delegates;
using UltraWebApplicationWithAssets1.HTML.Pages.FromAssets;
using ScriptCoreLib.Delegates;
using ScriptCoreLib.Ultra;

namespace UltraWebApplicationWithAssets1
{
	public sealed class UltraApplication
	{
		public UltraApplication(IHTMLElement e)
		{
			var a = new HTMLPage1();

			a.Button2.innerText = "Enumerate";

			Action Enumerate =
				delegate
				{
					a.Content.removeChildren();

					new UltraWebService().Enumerate(
						(Key, Field1) =>
						{
							var Delete = new IHTMLButton("Delete");
							var Item =
								new IHTMLDiv
								{
									Delete,
									Key + " " + Field1
								};

							Item.AttachTo(a.Content);

							ImplementDelete(Key, Delete, Item);
						}
					);
				};

			a.Button1.innerText = "Add";


			a.Button1.onclick +=
				delegate
				{
					new UltraWebService().Add(
						"Click at client time: " + DateTime.Now,
						delegate
						{
							Enumerate();
						}
					);

				};

			a.Button2.onclick +=
				delegate
				{
					a.Button2.style.color = "";
					Enumerate();
				};

			a.Container.AttachToDocument();
		}

		private static void ImplementDelete(string Key, IHTMLButton Delete, IHTMLDiv Item)
		{
			Delete.onclick +=
				delegate
				{
					Delete.disabled = true;

					new UltraWebService().Delete(
						Key,
						delegate
						{
							Item.Orphanize();
						}
					);

				};
		}

	}

	class vNext1
	{
		private static void ImplementDelete(string Key, IHTMLButton Delete, IHTMLDiv Item)
		{
			Delete.onclick +=
				delegate
				{
					Delete.disabled = true;

					Async<string, string> ServerDelete =
						(Key__, Yield) =>
						{
							Tier.Server();

							using (var ctx = new DataClasses1DataContext())
							{

								ctx.Table1s.DeleteAllOnSubmit(
									ctx.Table1s.Where(k => k.Key == Convert.ToInt32(Key__))
								);

								ctx.SubmitChanges();
							}

							if (Yield != null)
								Yield("");
						};

					ServerDelete(Key,
						delegate
						{
						}
					);


				};
		}


	}

	class vNext2
	{
		private static void ImplementDelete(string Key, IHTMLButton Delete, IHTMLDiv Item)
		{
			Delete.onclick +=
				delegate
				{
					Delete.disabled = true;

					Async<string, string> ServerDelete =
						(Key__, Yield) =>
						{
							new CodeAtServer().Invoke();

							using (var ctx = new DataClasses1DataContext())
							{

								ctx.Table1s.DeleteAllOnSubmit(
									ctx.Table1s.Where(k => k.Key == Convert.ToInt32(Key__))
								);

								ctx.SubmitChanges();
							}

							if (Yield != null)
								Yield("");
						};

					ServerDelete(Key,
						delegate
						{
						}
					);


				};
		}


	}

	[Tier(TierEnum.Server)]
	class CodeAtServer
	{

		internal void Invoke()
		{
			throw new NotImplementedException();
		}
	}

	class vNext3
	{
		private static void ImplementDelete(string Key, IHTMLButton Delete, IHTMLDiv Item)
		{
			Delete.onclick +=
				delegate
				{
					Delete.disabled = true;

					Tier.Server();

					using (var ctx = new DataClasses1DataContext())
					{

						ctx.Table1s.DeleteAllOnSubmit(
							ctx.Table1s.Where(k => k.Key == Convert.ToInt32(Key))
						);

						ctx.SubmitChanges();
					}

					Tier.JavaScript();

					//...

					Tier.Flash();

					//...

				};
		}


	}
}

File: templates/TwentyTen/VisualBasic/UltraWebApplicationWithDLinq/UltraWebApplicationWithDLinq/UltraApplication.vb
Imports ScriptCoreLib.JavaScript.DOM.HTML
Imports ScriptCoreLib.JavaScript.Extensions
Imports ScriptCoreLib.JavaScript

Imports UltraWebApplicationWithDLinq.HTML.Pages

Public NotInheritable Class UltraApplication

    Dim a As IHTMLPage1

    Dim WithEvents button1 As New IHTMLButton
    Dim WithEvents button2 As New IHTMLButton

    Public Sub New(ByVal a As IHTMLPage1)

        Me.a = a


        Native.Document.title = "UltraWebApplicationWithDLinq"

        Me.button1 = a.Button1
        Me.button2 = a.Button2

        a.Button1.innerText = "Add"
        a.Button2.innerText = "Enumerate"

    End Sub




    Private Sub button1_onclick(ByVal e As ScriptCoreLib.JavaScript.DOM.IEvent) Handles button1.onclick

        button2.style.color = "red"

        Dim w As New UltraWebService

        w.Add("client time: " & DateTime.Now, AddressOf Enumerate)

    End Sub

    Sub YieldEnumerate(ByVal ID As String, ByVal Text1 As String)
        Dim x As New IHTMLDiv

        x.innerText = Text1

        x.AttachTo(a.Content)
    End Sub

    Sub Enumerate(ByVal dummy As String)
        a.Content.Clear()

        Dim w As New UltraWebService
        button2.style.color = ""

        w.Enumerate(
            Sub(Data As XElement)

                a.Content.Add(
                    <div>
                        <code>
                            <span>#</span><span><%= Data.<ID>(0).Value %></span>
                            <span> - </span>
                            <span><%= Data.<Text1>(0).Value %></span>
                        </code>
                    </div>
                )
            End Sub
        )

    End Sub
    Private Sub button2_onclick(ByVal e As ScriptCoreLib.JavaScript.DOM.IEvent) Handles button2.onclick


        Enumerate("")

    End Sub
End Class
